actions/event-parsers.js

```diff
@@ -82,25 +104,24 @@ module.exports = {
   tpch: (body, v) => {
     let $ = cheerio.load(body);
     let events = [];
+    const month = moment().format('MMMM');
 
-    $('#blog li').first().find('p').each((i,e) => {
-      let date;
-      let artist;
+    $(`h1:contains(${month})`).parent('header').parent('article').find('p').each((i,e) => {
       const listing = $(e).text();
-      const listingArr = listing.split('~');
+      const [ day, artist ]= listing.split('~');
+      let date = moment(`${month} ${day}`, 'MMMM ddd Do');
 
-      if (listingArr.length > 1) {
-        date = listingArr[0].trim();
-        date += ' 7pm (Sunday 1pm)';
-        artist = listingArr[1].trim();
-        events.push({ date, artist });
-      }
-      else if (listingArr.length === 1) {
-        const tmpVar = listingArr[0].trim();
-        if (tmpVar != '') {
-          date = artist = tmpVar;
-          events.push({ date, artist });
-        }
+      if (date.isValid() && dateData.isThisWeek(date)) {
+        const dayName = date.format('dddd');
+        const eventTime = dayName === 'Sunday' ? '1:00 PM' : '7:00 PM'
```

Comment:
All of their events start at 7pm unless it is Sunday, in which it starts at 7pm. I could add that logic in when I am instantiating the moment date object, but in that case, I would need to suss  it out of the string which always varies in one weird way or another. Check out the format here: http://peekskillcoffee.com/events

This is the place that is run by my bass player's wife, the one that we always have to hassle each month to even put out their events. Their formatting of each event is always a smorgasbord of formatting and such. :/